### PR TITLE
Refatoring interval

### LINKED ISSUE #3: Make Interval implement Stream
It's currently pretty inconvenient to use `Interval`, as you need to keep re-assigning the `Interval` on completion. It'd be nicer if `Interval` were a stream that yielded an empty type each time the timer expired. Or perhaps both APIs could be supported?

### PR DESCRIPTION
**Pull request by DoumanAsh**
Fixes #3

### CODE REVIEW

File: src/interval.rs [hunk 1]
```diff
@@ -53,28 +50,53 @@ impl Interval {
...
-                    timer.restart(&interval, ctx.waker());
-                    task::Poll::Ready(Interval::Ongoing(timer, interval))
-                },
-                task::Poll::Pending => {
-                    *self = Interval::Ongoing(timer, interval);
-                    task::Poll::Pending
-                },
+        match Future::poll(Pin::new(&mut self.timer), ctx) {
+            task::Poll::Ready(()) => {
+                self.restart(ctx);
```
Comment (jonhoo) [inline src/interval.rs]: I think in general the policy is for `Future` to never poll again `Ready` has been returned. This is the reason things like `FusedFuture` exist. Even though it makes for a nice API here, I think we may want to keep it the way it currently is for the `Future` impl (the `Stream` can be the way to do it with `&mut`)

Comment (DoumanAsh) [reply src/interval.rs]: Yeah, but there is no requirements on `Future::poll` behavior so I want to keep this nice API to be able to do something like `interval.as_mut().await`

**Comment by jonhoo** [comment]
@DoumanAsh Should `stream` be a default feature?

**Comment by DoumanAsh** [comment]
No, by default I prefer to keep minimum here

**Pull request merged**